MetroscopeModelingLibrary/RefMoistAir/Machines/AirCompressor.mo

```diff
@@ -0,0 +1,134 @@
+within MetroscopeModelingLibrary.RefMoistAir.Machines;
+model AirCompressor
+
+  extends MetroscopeModelingLibrary.Partial.BaseClasses.FlowModel(
+    redeclare MetroscopeModelingLibrary.RefMoistAir.Connectors.Inlet C_in,
+    redeclare MetroscopeModelingLibrary.RefMoistAir.Connectors.Outlet C_out,
+    redeclare package Medium = RefMoistAirMedium,
+    Q_0 = 500, rho_0 = 1) annotation (IconMap(primitivesVisible=false));
+
+  package RefMoistAirMedium = MetroscopeModelingLibrary.Utilities.Media.RefMoistAirMedium;
+
+  import MetroscopeModelingLibrary.Utilities.Units;
+  import MetroscopeModelingLibrary.Utilities.Units.Inputs;
+
+  Inputs.InputReal tau(start=15, min = 1) "Compression rate";
+  Inputs.InputReal eta_is(start=0.8, min=0, max=1) "Nominal isentropic efficiency";
+
+  Units.SpecificEnthalpy h_is(start=1e6) "Isentropic compression outlet enthalpy";
+  RefMoistAirMedium.ThermodynamicState state_is "Isentropic compression outlet thermodynamic state";
+  Real Q_reduced "Compressor reduced mass flow";
+
+  // Failure modes
+  parameter Boolean faulty = false;
+  Units.Percentage eta_is_decrease(min = 0, max=100) "percentage decrease of eta_is";
+  Units.Percentage tau_decrease(min = 0, max=100) "percentage decrease of tau";
+
+  // Indicators
+  Units.MassFraction x_liq_in; // Liquid water mass fraction at the inlet
+  Units.MassFraction x_liq_out; // Liquid water mass fraction at the outlet
+  Real relative_humidity_in; // Relative humidity at the inlet
+  Real relative_humidity_out; // Relative humidity at the outlet
```

Comment:
I think it's better to declare this in the flow model, like that we have access to the relative_humidity in all the components of RefMoistAir

Reply:
I will do that in a next PR